apps/src/blockly/addons/cdoSerializationHelpers.js

```diff
@@ -162,3 +161,23 @@ function addUnusedFrame(block) {
     block.unusedSvg_.render(block.svgGroup_, block.RTL);
   }
 }
+
+/**
+ * Reorders an array of blocks based on the given blockOrderMap.
+ * If the blockOrderMap is invalid (null or size mismatch), returns the original array.
+ *
+ * @param {Array} blocks - The array of blocks to be reordered.
+ * @param {Map} blockOrderMap - A map with block index as key and the desired order index as value.
+ * @returns {Array} The reordered array of blocks or the original array if blockOrderMap is invalid.
+ */
+function reorderBlocks(blocks, blockOrderMap) {
+  if (!blockOrderMap || blockOrderMap.size !== blocks.length) {
+    return blocks;
+  }
+  const orderedBlocks = new Array(blocks.length);
+  blocks.forEach((block, index) => {
+    orderedBlocks[blockOrderMap.get(index)] = block;
```

Comment:
I have not used `Map`s much yet. For learning purposes, is there a particular thing that's useful about using one here?

Reply:
I also haven't used them, but thought it would be interesting to try here. I tried this approach because I initially wasn't sure how I'd end up iterating through to reorder the blocks. (In my head, I found myself literally saying that each block needed to be able to "map" back to its original index.) 
Looking at what I have here, I think it probably could have just been an array (e.g. `[2, 0, 1]`) since each value also inherently has an index in the array. A map would probably be more justified if the keys aren't sequential (or aren't just integers). Now I'm wondering if it's worth refactoring this.

I also wonder if there's any argument to be made about the readability of each structure, ie:

Array:
```
[
    2,
    0,
    1
]
```
Map:
```
[
    {
        "key": 0,
        "value": 2
    },
    {
        "key": 1,
        "value": 0
    },
    {
        "key": 2,
        "value": 1
    }
]
```

Reply:
A map is great here, and there's no need to change. I would've defaulted to using a vanilla object (i.e., `{ 0: 2, 1: 0, 2: 1 }`), which is why I asked!